pkg/repository/repository.go

```diff
@@ -65,7 +65,7 @@ func getURLFromPath(path string) repositoryURL {
 func NewRepository(path string) Repository {
 	repoURL := getURLFromPath(path)
 
-	var controller repositoryInterface = newGitRepoController()
+	var controller repositoryInterface = newGitRepoController(path)
```

Comment:
withput passing `path` to the repo controller, when we check if the repo is clean, we check against an empty path and the result is always false (dirty)